File: cgi-bin/env.py
#!/usr/bin/env python3

import os, datetime, sys, urllib.parse

try:
    meth = os.environ['REQUEST_METHOD']
except:
    meth = 'None'

print("Content-Type: text/html")
print("Pragma: no-cache")
print("");

print("""<!DOCTYPE HTML>
<html>
<head>
<meta charset='utf-8'>
<title>Тест</title>
</head>
<body>
""")

list_env = os.environ
for i in list_env:
    print('<b>{}: {}</b><br>'.format(i, list_env[i]))

if meth == 'POST':
    post_data = urllib.parse.parse_qsl(sys.stdin.read())
    print('<p>stdin: {}</p>'.format(post_data))
    for i in post_data:
        print('<b>{}: {}</b><br>'.format(i[0], i[1]))
elif meth == 'GET':
    get_data = urllib.parse.parse_qsl(os.environ['QUERY_STRING'])
    print('<p>{}</p>'.format(get_data))
    for i in get_data:
        print('<b>{}: {}</b><br>'.format(i[0], i[1]))

now = datetime.datetime.utcnow()
print("""<form action=\"env.py\" method=\"%s\">
<input type=\"hidden\" name=\"par1\" value=\".-./. .+.!.?.,.~.#.&.>.<.^.\">
<p><input type=\"submit\" value=\"Get ENV\"></p>
</form>
<hr>
%s
</body>
</html>""" %(meth, now.strftime("%a, %d %b %Y %H:%M:%S GMT")))

File: cgi-bin/env.pl
#!/usr/bin/perl

use strict;
use warnings;
use POSIX;
use File::Basename;
use URI::Escape;

my $time = localtime;

my $meth = $ENV{'REQUEST_METHOD'};
if (!defined($meth))
{
    $meth = '------';
}

my $data = '';
if (($meth =~ /POST/))
{
    $data = uri_unescape(<STDIN>);
#    my $length = $ENV{CONTENT_LENGTH};
#    read STDIN, $data, $length;
#    $data = uri_unescape($data);
}

print "Content-type: text/html; charset=utf-8\n\n";
print "<!DOCTYPE html>
<html>
<head>
<title>Environment Dumper </title>
</head>
<body>
<center>
<table border=1>\n";
foreach (sort keys %ENV)
{
    print "<tr><td>$_</td><td>$ENV{$_}</td></tr>\n";
}
print "</table>
<p>$data</p>
<form action=\"env.pl\" method=\"$meth\">
<input type=\"hidden\" name=\"name\" value=\".-./. .+.!.?.,.~.#.&.>.<.^.\">
<input type=\"submit\" value=\"Get ENV\">
</form>
</body>
</html>";

